src/theory/uf/theory_uf_rewriter.cpp

```diff
@@ -125,28 +93,21 @@ RewriteResponse TheoryUfRewriter::postRewrite(TNode node)
             << "uf-ho-beta : ....new lambda : " << new_body << "\n";
       }
 
-      // build capture-avoiding substitution since in HOL shadowing may have
-      // been introduced
-      if (d_isHigherOrder)
-      {
-        Node arg = node[1];
-        Node var = lambda[0][0];
-        new_body = expr::substituteCaptureAvoiding(new_body, var, arg);
-      }
-      else
-      {
-        TNode arg = node[1];
-        TNode var = lambda[0][0];
-        new_body = new_body.substitute(var, arg);
-      }
+      TNode arg = node[1];
+      TNode var = lambda[0][0];
+      new_body = new_body.substitute(var, arg);
+
       Trace("uf-ho-beta") << "uf-ho-beta : ..new body : " << new_body << "\n";
       return RewriteResponse(REWRITE_AGAIN_FULL, new_body);
     }
   }
   else if (k == kind::LAMBDA)
   {
     Node ret = rewriteLambda(node);
-    return RewriteResponse(REWRITE_DONE, ret);
+    if (ret != node)
+    {
+      return RewriteResponse(REWRITE_AGAIN_FULL, ret);
+    }
```

Comment:
Why is this necessary? What new rewrites do we expect to apply once shadowing is eliminated?

Reply:
None, I guess. In general `REWRITE_AGAIN_FULL` is a way to be safe, else this is a small optimization.